### DISCUSSION TITLE
howto containerize executable python

### DISCUSSION THREAD

**Discussion by markat1** [microsoft/aspire#5722] [Q&A] [answered]
If I want to convert a executable python into a containerized python, how would I do that?

I have tried the following without any luck.

added a Dockerfile into a python project.

![image](https://github.com/user-attachments/assets/26666baa-fbfd-4d35-a657-3f3df8203a4c)

then I tried adding a PublishAsDockerFile(), which I thought would call the Dockerfile inside the python project and spin up a new container.

It  didn't spin up any container, unfortunately:

![image](https://github.com/user-attachments/assets/5224b25a-ba90-4ed1-8cb4-2efa990e5faa)

**Comment by davidfowl** [comment, marked as answer]
If you want to run locally with a docker file then you can use AddDockerfile instead

**Reply by markat1** [reply]
thanks @davidfowl 

PublishAsDockerFile() is only for deployment?

Is it a must that I add PublishAsDockerfile on python executables, when I deploy?

**Reply by davidfowl** [reply]
> PublishAsDockerFile() is only for deployment?

That's what the Publish* prefix means. Only do this action when publishing.

> Is it a must that I add PublishAsDockerfile on python executables, when I deploy?

If you use AddDockerFile then you don't need to.
